src/main/java/org/candlepin/pki/certs/SCACertificateGenerator.java

```diff
@@ -296,7 +296,12 @@ private Content createContent(Owner owner, Environment environment) {
             path += "/" + Util.encodeUrl(owner.getKey());
         }
         if (environment != null) {
-            path += "/" + Util.encodeUrl(environment.getName());
+            String environmentName = environment.getName();
+            if (environmentName != null) {
+                environmentName = environmentName.replaceAll("[^a-zA-Z0-9._-]", "");
```

Comment:
Are we certain removing the character is the right choice here? Seems like the intent is to have the slash recognized as a path segment separator.

Reply:
Good call! Verified that the slash in the environment name should not be removed and recognized as a path segment separator.